File: src/main/java/tomekkup/helenos/types/qx/query/CqlQuery.java
package tomekkup.helenos.types.qx.query;

/**
 * ********************************************************
 * Copyright: 2012 Tomek Kuprowski
 *
 * License: GPLv2: http://www.gnu.org/licences/gpl.html
 *
 * @author Tomek Kuprowski (tomekkuprowski at gmail dot com)
 * *******************************************************
 */
public class CqlQuery<K,N,V> extends AbstractQuery<K,N,V> {

    private String query;

    public CqlQuery() {
        super();
    }

    public CqlQuery(Class<K> keyClass, Class<N> nameClass, String keyspace, String columnFamily, String query) {
        super(keyClass, nameClass, keyspace, columnFamily);
        this.query = query;
    }

    public String getQuery() {
        return query;
    }

    public void setQuery(String query) {
        this.query = query;
    }
}

File: src/main/frontend/source/class/helenos/model/CqlQuery.js
/* ************************************************************************
Copyright:
  2012 Tomek Kuprowski
License:
  GPLv2: http://www.gnu.org/licences/gpl.html
Authors:
  Tomek Kuprowski (tomekkuprowski at gmail dot com)
 ************************************************************************ */
qx.Class.define('helenos.model.CqlQuery', {
    extend : helenos.model.AbstractQuery,
    
    construct : function(cfDef, consistencyLevel)
    {
        this.base(arguments, cfDef, consistencyLevel);
    },

    properties :
    {
        query : {
            check : 'String'
        }
    }
});
